### PR TITLE
feat: add support for Rails 7.1

### PR DESCRIPTION
**Pull request by hoshy**
The changes from https://github.com/rails-on-services/apartment/pull/233 are included here.

### CODE REVIEW

File: lib/apartment/active_record/schema_migration.rb [hunk 1]
```diff
@@ -1,11 +1,23 @@
 # frozen_string_literal: true
+#
+require 'active_record/schema_migration'
 
 module ActiveRecord
-  class SchemaMigration < ActiveRecord::Base # :nodoc:
-    class << self
-      def table_exists?
-        connection.table_exists?(table_name)
+  if ActiveRecord::SchemaMigration < ActiveRecord::Base
```
Comment (m-zielinski) [inline lib/apartment/active_record/schema_migration.rb:6]: suggestion: I think this condition is unneccessary, as opening an existing class without the parent class name doesn't change the inheritance, so this:
```
class SchemaMigration
```
should work no matter the parent.

**Comment by davidedisomma1** [comment]
is there an update on this one? would be nice to have

**Comment by romarioliveira25** [comment]
Any new about that?

File: lib/apartment/active_record/schema_migration.rb [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
+require 'active_record/schema_migration'
+
 module ActiveRecord
-  class SchemaMigration < ActiveRecord::Base # :nodoc:
+  class SchemaMigration # :nodoc:
     class << self
       def table_exists?
         connection.table_exists?(table_name)
       end
+      alias :table_name :name
```
Comment (eraffel-MDSol) [inline lib/apartment/active_record/schema_migration.rb:11]: Why this change? I think this is causing problems

**Review by eraffel-MDSol — COMMENTED** [review]
It's a bit hard to follow some of the changes in this PR, because they're not accompanied by tests or comments explaining their neccessity. I opened my own PR, but I'm worried that you found stuff in your testing that I didn't run into

**Comment by hoshy** [comment]
> It's a bit hard to follow some of the changes in this PR, because they're not accompanied by tests or comments explaining their neccessity. I opened my own PR, but I'm worried that you found stuff in your testing that I didn't run into

Let's continue with yours. But yes, needs some (manual) testing.

**Comment by hoshy** [comment]
closing due to https://github.com/rails-on-services/apartment/pull/243
